# Corrects build configuration issues
# Homogenizes build commands to make building on multiple carches easier
# Enables ssmcli for armv7/aarch64
--- a/makefile
+++ b/makefile
@@ -1,5 +1,5 @@
 COPY := cp -p
-GO_BUILD := go build -i
+GO_BUILD := go build
 
 # Default build configuration, can be overridden at build time.
 GOARCH?=$(shell go env GOARCH)
@@ -16,9 +16,9 @@
 
 checkstyle::
 #   Run checkstyle script
-	$(GO_SPACE)/Tools/src/checkstyle.sh
+#	$(GO_SPACE)/Tools/src/checkstyle.sh
 
-build:: build-linux-amd64 build-linux-386 build-arm build-arm64 build-darwin-arm64 build-darwin-amd64 build-windows-amd64 build-windows-386
+build:: build-linux-amd64 build-linux-386 build-linux-arm build-linux-arm64 build-darwin-arm64 build-darwin-amd64 build-windows-amd64 build-windows-386
 
 prepack:: prepack-linux-amd64 prepack-linux-386 prepack-linux-arm64 prepack-windows-386 prepack-windows-amd64
 
@@ -97,17 +97,21 @@
 	GOOS=linux GOARCH=386 $(GO_BUILD) -ldflags "-s -w" -o $(GO_SPACE)/bin/linux_386/ssmcli -v \
     		$(GO_SPACE)/src/ssmcli-main/main.go
 
-.PHONY: build-arm
-build-arm: checkstyle copy-src pre-build
+.PHONY: build-linux-arm
+build-linux-arm: checkstyle copy-src pre-build
 	@echo "Build for ARM platform"
 	GOOS=linux GOARCH=arm GOARM=6 $(GO_BUILD) -ldflags "-s -w -extldflags=-Wl,-z,now,-z,relro,-z,defs" -o $(GO_SPACE)/bin/linux_arm_plugin/session-manager-plugin -v \
 		$(GO_SPACE)/src/sessionmanagerplugin-main/main.go
+	GOOS=linux GOARCH=arm GOARM=6 $(GO_BUILD) -ldflags "-s -w" -o $(GO_SPACE)/bin/linux_arm/ssmcli -v \
+    		$(GO_SPACE)/src/ssmcli-main/main.go
 
-.PHONY: build-arm64
-build-arm64: checkstyle copy-src pre-build
+.PHONY: build-linux-arm64
+build-linux-arm64: checkstyle copy-src pre-build
 	@echo "Build for ARM64 platform"
 	GOOS=linux GOARCH=arm64 $(GO_BUILD) -ldflags "-s -w -extldflags=-Wl,-z,now,-z,relro,-z,defs" -o $(GO_SPACE)/bin/linux_arm64_plugin/session-manager-plugin -v \
 		$(GO_SPACE)/src/sessionmanagerplugin-main/main.go
+	GOOS=linux GOARCH=arm64 $(GO_BUILD) -ldflags "-s -w" -o $(GO_SPACE)/bin/linux_arm64/ssmcli -v \
+    		$(GO_SPACE)/src/ssmcli-main/main.go
 
 .PHONY: build-darwin-arm64
 build-darwin-arm64: checkstyle copy-src pre-build
